salt/dex/dex.yaml

```diff
@@ -0,0 +1,113 @@
+---
+kind: ConfigMap
+apiVersion: v1
+metadata:
+  name: dex
+  namespace: kube-system
+data:
+  config.yaml: |
+    issuer: "https://{{ grains['caasp_fqdn'] }}:32000"
+    storage:
+      type: kubernetes
+      config:
+        inCluster: true
+    web:
+      https: 0.0.0.0:5556
+      tlsCert: /etc/dex/tls/dex.crt
+      tlsKey: /etc/dex/tls/dex.key
+    connectors:
+    - type: ldap
+      id: ldap
+      name: LDAP
+      config:
+        host: "{{ pillar['dashboard'] }}"
+        startTLS: true
+        bindDN: cn=admin,{{ pillar['ldap_internal_infra_domain'] }}
+        bindPW: admin
+        rootCA: {{ pillar['ssl']['ca_file'] }}
+        userSearch:
+          baseDN: ou=People,{{ pillar['ldap_internal_infra_domain'] }}
+          filter: "(objectClass=inetOrgPerson)"
+          username: mail
+          idAttr: uid
+          emailAttr: mail
+          nameAttr: gecos          
+        groupSearch:
+          baseDN: ou=Groups,{{ pillar['ldap_internal_infra_domain'] }}
+          filter: "(objectClass=posixGroup)"
+
+          userAttr: uid
+          groupAttr: memberUid
+
+          nameAttr: cn       
+
+    oauth2:
+      skipApprovalScreen: true
+
+    staticClients:
+    - id: caasp-cli
+      redirectURIs:
+      - 'http://127.0.0.1'
+      name: "CaaSP CLI"
+      secret: "swac7qakes7AvucH8bRucucH"
```

Comment:
This is a tough one. The CLI needs the exact same secret to perform the OAuth2 handshake. So, if we generate it, everyone who uses the CLI will need it in a local file.